### ISSUE TITLE
Update support phone (and maybe other fields): Stripe API error is not exposed to merchants

### ISSUE THREAD

**Issue by htdat** [Automattic/woocommerce-payments#5516]
### Describe the bug

Coming from this PR review: https://github.com/Automattic/woocommerce-payments/pull/5471#pullrequestreview-1279158193

Update support phone: Stripe API error is not exposed to the front-end.

### To Reproduce

1. Go to wp-admin > Payments > Settings, or wp-admin > Payments > Card Readers > Receipt Details (- in your local server, return `true` for `has_card_readers_available` flag/method, like this 2ef2d-pb) 
2. Try to use a US store for this setting.
4. Set a valid number outside US, e.g. `+441332412251` for GB, and save.
5. It appears that the setting is saved successfully. 

Check logs at Stripe Dashboard https://dashboard.stripe.com/acct_xxxxxxx/test/logs, get an error: 

```
{
  "error": {
    "message": "\"+441332412251\" is not a valid phone number",
    "param": "business_profile[support_phone]",
    "request_log_url": "https://dashboard.stripe.com/acct_xxxxxx/test/logs/req_oYcgjWfANtAQRb?t=1675432902",
    "type": "invalid_request_error"
  }
}
```

### Actual behavior

The settings seem to be saved successfully but they are not actually saved due to an error from Stripe API.

### Expected behavior

Make sure that settings are really saved or relay error message to the front-end.

### Additional context

- From what I debugged, `WC_Payment_Gateway_WCPay::update_account` use `add_error` to display admin error notices in the old way rather than expose the error message from Stripe API to the rest request. 
- We may also enhance our validation before passing a phone parameter to Stripe API.

**Comment by htdat** [comment]
From my observation, it seems that Stripe only allows phone numbers respective to the country of the store.

Trying to confirm: p1675435995673419-slack-C9976E5MJ

_Update Mar 10_: I am still verifying this on the convo link above.

**Comment by htdat** [comment]
> From my observation, it seems that Stripe only allows phone numbers respective to the country of the store.

This still needs to be verified, however, the main issue here that we're not exposing errors from Stripe is still there. We would look into this though it's not a high priority of this issue.

**Comment by htdat** [comment]
Last update: p1679191211791349/1675435995.673419-slack-C9976E5MJ

> While there is not a documented list of country/ phone number exceptions/ validations we have... The team did clarify that US/ CA might have specific validations that might not apply to other regions.

I think whatever our validations are, we might not able to catch up with all errors from Stripe. So we should handle and relay Stripe API errors to merchants when they provide invalid inputs (from Stripe perspective).

**Comment by jessy-p** [comment]
This issue impacts Payments settings page, so assigning to Moltres (based on team responsibilities Pc2DNy-3z-p2) @daquinons. Assigning as part of Gamma Triage process PcreKM-yM-p2.

**Comment by anu-rock** [comment]
@vbelolapotkov Need your help in triaging this. The issue is related to IPP (not the Settings page as originally labeled), so I have relabeled it to be re-triaged.

**Comment by anu-rock** [comment]
Moving it back to Moltres backlog as this issue is related to account settings. p1713541230042679-slack-C02L752DS0G

**Comment by dpaun1985** [comment]
It seems the problem is no longer occurring.
I was able to save a UK phone number for a US store from the  `Payments -> Settings` section and from `Payments -> Card Readers -> Receipt Details`.
I checked the server logs, but I couldn't find any errors related to the phone number.

<img width="1122" alt="Image" src="https://github.com/user-attachments/assets/d9f3792d-da87-440f-8680-a22bbe0236db" />

I will close the issue for now,  if it will be reproduced in the future we can reopen it.
